pyccel/ast/builtins.py

```diff
@@ -881,9 +881,10 @@ def print_string(self):
         can be used in a print  statement
         """
         prec = self.precision
+        str_dtype = str(self.dtype)
         return LiteralString("<class '{dtype}{precision}'>".format(
-            dtype = str(self.dtype),
-            precision = '' if prec in (None, -1) else prec*8))
+            dtype = str_dtype,
+            precision = '' if prec in (None, -1) else (prec * 8 * (2 if str_dtype=="complex" else 1))))
```

Comment:
I think it would be safer to check if `dtype is NativeComplex()` in case some wise guy in the future decides to name his class "complex"

Reply:
fixed in https://github.com/pyccel/pyccel/pull/1167/commits/f2bc151f02f353d1d6ea6060edd3a641f5d3f669